### PR TITLE
Adding networking changes (Eventgrid to SB enablement)

### LINKED ISSUE #1841: Airlock processor handles request approval
As an Airlock processor in TRE
I want to be able to be able to consume and operate on the approval event 
So that requests approval are handled

**Acceptance Criteria**
- [ ] Data artifact are no longer available in `In-progress storage` (which is handled by #1963  )
- [ ] Data artifact are available in `Approved storage`
- [ ] Step result event is sent by the airlock processor when the data copy is done
- [ ] Data movement is logged for auditing purpose
- [ ] Data Steward can no longer access the `In-progress storage`
- [ ] A notification event with all the details sent to Event grid (to be consumed by #1964 )

### LINKED ISSUE #1842: Airlock processor handles request rejection
As an Airlock processor in TRE
I want to be able to be able to consume and operate on rejection event 
So that requests rejections are handled

**Acceptance Criteria**
- [ ] Data artifact are copied to the 'rejected' containers (import rejects or export rejected)
- [ ] Step result event is sent by the airlock processor when the data copy is done
- [ ] Data deletion event should be sent (which is handled by #1963 )
- [ ] Data Steward can no longer access the `In-progress storage`
- [ ] A notification event with all the details sent to Event grid (to be consumed by #1964 )

### LINKED ISSUE #2099: Event grid can't send messages over PE to SB
Event grid can't send messages over PE to SB.

As a TRE Developer
I want to make sure the messages from the EG are arriving to SB
So that the flow functions

## Acceptance criteria

- [ ] messages arrives from the event grid topics to the service bus queues/topics

### PR DESCRIPTION
**Pull request by eladiw**
Fixes #2099  

1. Update networking configuration:
  * Each airlock topic/system topic gets a system-assigned identity (user-assigned don't work for this purpose of enabling secured messages transfer for EG to SB)
3. Each of the system assigned identities gets a SB Data Sender role assignment for the SB
4. SB accepts connections from selected networks + allowed trusted MS services (EG is one of them)

### CODE REVIEW

**Review by LizaShak — APPROVED** [review]

File: airlock_processor/StatusChangedQueueTrigger/__init__.py [hunk 1]
```diff
@@ -100,25 +100,33 @@ def get_source_dest_env_vars(new_status: str, request_type: str, short_workspace
         dest_account_name = constants.STORAGE_ACCOUNT_NAME_IMPORT_INPROGRESS.format(tre_id)
         source_account_rg = constants.CORE_RG_NAME.format(tre_id)
         dest_account_rg = source_account_rg
-        logging.info("source account [%s rg: %s]. dest account [%s rg: %s]", source_account_name, source_account_rg, dest_account_name, dest_account_rg)
     elif new_status == 'submitted' and request_type == 'export':
         source_account_name = constants.STORAGE_ACCOUNT_NAME_EXPORT_INTERNAL.format(short_workspace_id)
         dest_account_name = constants.STORAGE_ACCOUNT_NAME_EXPORT_INPROGRESS.format(short_workspace_id)
-        source_account_rg = constants.CORE_RG_NAME.format(tre_id, short_workspace_id)
+        source_account_rg = constants.WS_RG_NAME.format(tre_id, short_workspace_id)
```
Comment (tamirkamara) [inline airlock_processor/StatusChangedQueueTrigger/__init__.py:106]: nit: reduce the shorthand here...
You have long names for storage accounts so my not here - WORKSPACE_RESOURCE_GROUP_NAME?

**Review by tamirkamara — APPROVED** [review]

**Pull request merged**